src/signals.c

```diff
@@ -302,39 +302,58 @@ static RETSIGTYPE sig_terminate(int signo) {
   term_signo = signo;
 
   if (signo == SIGSEGV ||
-      signo == SIGXCPU
 #ifdef SIGBUS
-      || signo == SIGBUS) {
-#else
-     ) {
+      signo == SIGBUS ||
 #endif /* SIGBUS */
+      signo == SIGILL ||
+      signo == SIGINT ||
+      signo == SIGXCPU) {
+    const char *signame = "(unsupported)";
+
+    switch (signo) {
+      case SIGSEGV:
+        recvd_signal_flags |= RECEIVED_SIG_SEGV;
+        signame = "SIGSEGV";
+        break;
+
+      case SIGXCPU:
+        recvd_signal_flags |= RECEIVED_SIG_XCPU;
+        signame = "SIGXCPU";
+        break;
+
+      default:
+        recvd_signal_flags |= RECEIVED_SIG_TERMINATE;
```

Comment:
with this you are ignoring SIGTERM :) this means, that childs will never be killed, because of first bad condition.